src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```
@@ -0,0 +1,194 @@
+            selectStatement.setLong(1, id);
+            ResultSet resultSet = selectStatement.executeQuery();
+            Car car = null;
+            if (resultSet.next()) {
+                car = getCarFromResultSet(resultSet);
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> cars = new ArrayList<>();
+        String query = "SELECT c.id, c.manufacturer_id, c.model, m.name, m.country "
+                + "FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement selectStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = selectStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars from DB", e);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement updateStatement = connection.prepareStatement(query)) {
+            updateStatement.setLong(1, car.getManufacturer().getId());
+            updateStatement.setString(2, car.getModel());
+            updateStatement.setLong(3, car.getId());
+            deleteDrivers(car.getId());
+            addDrivers(car);
+            return updateStatement.executeUpdate() > 0 ? car : null;
```

Maybe it's better to throw here exception in that case instead of passing null value